docs/NEW_LANGUAGE_FEATURE_INTEGRATION_PROCESS.md

```diff
@@ -0,0 +1,144 @@
+# New Language Feature Check Integration Process [WIP]
```

Comment:
@nrmancuso Do we consider module-info a new language feature? Using that as the worst example, it requires a whole new grammar parser and tree. Wasn't sure if this was suppose to cover things like that or we are just assuming when we come to this document, the grammar changes are done.

Reply:
> Wasn't sure if this was suppose to cover things like that or we are just assuming when we come to this document, the grammar changes are done.

Yes, we are assuming that the new syntax is already parsable by checkstyle. IMO, updating grammar and tree should be in a separate doc and we already have an issue for this in the project I hope @nrmancuso gets it done😄

Reply:
Then probably just make it straightforward and say this in the document so it is clear.

Reply:
Yeah, we can just extend the description here to say something like “after adding support for parsing…”

Reply:
>  Do we consider module-info a new language feature?

I am not sure, I would say that we can check the language feature by release openjdk page and see if it is there. Really, this is “bigger” than a language feature IMO.

Reply:
> Then probably just make it straightforward and say this in the document

done